parsec/CMakeLists.txt

```diff
@@ -234,11 +242,118 @@ if( BUILD_PARSEC )
   endif(CMAKE_Fortran_COMPILER_SUPPORTS_F90)
 
   install(TARGETS parsec
-    EXPORT parsec-targets
-    ARCHIVE DESTINATION ${PARSEC_INSTALL_LIBDIR}
-    LIBRARY DESTINATION ${PARSEC_INSTALL_LIBDIR}
-    PUBLIC_HEADER DESTINATION ${PARSEC_INSTALL_INCLUDEDIR})
+          EXPORT parsec-targets
+          ARCHIVE DESTINATION ${PARSEC_INSTALL_LIBDIR}
+          LIBRARY DESTINATION ${PARSEC_INSTALL_LIBDIR})
+  # Cmake < 3.20 does not support hierarchical PUBLIC and PRIVATE _HEADER,
+  # so we need to provide our own support by crafting a similar property
+  # PUBLIC_HEADER_H and PRIVATE_HEADER_H
+  # Extract these properties and properly install the files respecting the
+  # defined hierarchy
+  get_target_property(_public_headers parsec PUBLIC_HEADER_H)
+  foreach(_FILE ${_public_headers})
+    get_filename_component(_DIR ${_FILE} DIRECTORY)
+    INSTALL(FILES ${_FILE} DESTINATION ${PARSEC_INSTALL_INCLUDEDIR}/parsec/${_DIR})
+  endforeach()
+  get_target_property(_private_headers parsec PRIVATE_HEADER_H)
+  foreach(_FILE ${_private_headers})
+    get_filename_component(_DIR ${_FILE} DIRECTORY)
+    INSTALL(FILES ${_FILE} DESTINATION ${PARSEC_INSTALL_INCLUDEDIR}/parsec/${_DIR})
+  endforeach()
+
+  install(FILES utils/help-mca-param.txt
+          DESTINATION ${PARSEC_INSTALL_DATADIR}/parsec)
+  install(FILES
+          ${CMAKE_CURRENT_SOURCE_DIR}/runtime.h
```

Comment:
Why aren't these also dealt with `set_property(TARGET parsec PUBLIC_HEADER)`?

Reply:
With current cmake I could have used PUBLIC_HEADER with all headers going directly in include. It looks cleaner if the PUBLIC_HEADER_H is only used for headers outside the parsec/CMakeLists.txt, and everything else is installed via INSTALL.